### PR TITLE
[SPARK-41304][CONNECT][PYTHON][DOCS] Add missing docs for DataFrame API

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Add missing docs for DataFrame API.

### Why are the changes needed?

Improve documentation

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by amaliujia** [comment]
@zhengruifeng

File: python/pyspark/sql/connect/dataframe.py [hunk 1]
```diff
@@ -457,6 +519,31 @@ def sample(
...
+            Sample with replacement or not (default ``False``).
+        fraction : float
+            Fraction of rows to generate, range [0.0, 1.0].
+        seed : int, optional
+            Seed for sampling (default a random seed).
+
+        Returns
+        -------
+        :class:`DataFrame`
+            Sampled rows from given DataFrame.
```
Comment (bjornjorgensen) [inline python/pyspark/sql/connect/dataframe.py:538]: Sampled -> Sample

File: python/pyspark/sql/connect/dataframe.py [hunk 2]
```diff
@@ -409,9 +441,39 @@ def join(
...
     def limit(self, n: int) -> "DataFrame":
+        """Limits the result count to the number specified.
+
+        .. versionadded:: 3.4.0
+
+        Parameters
+        ----------
+        num : int
+            Number of records to return. Will return this number of records
+            or whataver number is available.
```
[the remark this answers is not in the document]
Comment (bjornjorgensen) [reply python/pyspark/sql/connect/dataframe.py:452]: What do you mean here? 
Can we explain the "whatever number" in a better way?

File: python/pyspark/sql/connect/dataframe.py (hunk 1 above)
Comment (amaliujia) [reply python/pyspark/sql/connect/dataframe.py:538]: I guess this is sampled which is `Rows are sampled from the given DataFrame`

File: python/pyspark/sql/connect/dataframe.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (amaliujia) [reply python/pyspark/sql/connect/dataframe.py:452]: I upgraded the wording here a bit.

**Review by xinrong-meng — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by zhengruifeng** [comment]
Late LGTM

File: python/pyspark/sql/connect/dataframe.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bjornjorgensen) [reply python/pyspark/sql/connect/dataframe.py]: @amaliujia thank you, much better

**Pull request closed without merging**